src/main/groovy/com/github/eerohele/SaxonXsltTask.groovy

```diff
@@ -725,34 +737,64 @@ class SaxonXsltTask extends DefaultTask {
             //
             // Anyway, this approach likely covers a large chunk of how people
             // would conceivably use this feature anyway.
-        } else if (project.files(this.options.input).size() == 1 && project.buildDir.getAbsolutePath() != this.options.output.getAbsolutePath()) {
-            this.options.output
+        } else if (project.files(this.inputFiles).size() == 1 && project.buildDir.getAbsolutePath() != this.outputDir.getAbsolutePath()) {
+            this.outputDir
         } else {
-            new File(this.options.output, getOutputFileName(file))
+            new File(this.outputDir, getOutputFileName(file))
         }
     }
 
-    @OutputFiles
-    FileCollection getOutputFiles() {
-        if (this.options.input != null) {
-            project.files(project.files(this.options.input).collect { getOutputFile(it) })
-        } else {
-            project.files(this.options.output)
-        }
+    @InputFiles
+    @SkipWhenEmpty
+    @PathSensitive(PathSensitivity.RELATIVE)
```

Comment:
I believe the `@PathSensitive` attribute is a fix to #29. That should be in a separate commit. Ideally, there'd be just one commit that fixes Gradle 7 compatibility issues and nothing else.